Dockerfile

```diff
@@ -1 +1,21 @@
+FROM golang:1.16 as cli-builder
+WORKDIR /go/src/github.com/mendersoftware/mender-cli
+ARG MENDER_CLI_VERSION
+RUN if [ -z "$MENDER_CLI_VERSION" ]; then export MENDER_CLI_VERSION=master; fi && \
+    git clone -b $MENDER_CLI_VERSION https://github.com/mendersoftware/mender-cli.git . && \
+    make get-deps && \
+    make build
+
+FROM golang:1.18 as artifact-builder
+WORKDIR /go/src/github.com/mendersoftware/mender-artifact
+ARG MENDER_ARTIFACT_VERSION
+RUN if [ -z "$MENDER_ARTIFACT_VERSION" ]; then export MENDER_ARTIFACT_VERSION=master; fi && \
+    git clone -b $MENDER_ARTIFACT_VERSION https://github.com/mendersoftware/mender-artifact.git . && \
+    apt-get update && \
+    apt-get install -y gcc gcc-mingw-w64 gcc-multilib musl-dev liblzma-dev libssl-dev && \
```

Comment:
We should use https://github.com/mendersoftware/mender-artifact/blob/master/deb-requirements.txt for the dependencies instead of hard-coding them here.